assets/src/block-editor/helpers/index.js

```diff
@@ -272,237 +170,61 @@ export const filterBlocksEdit = ( BlockEdit ) => {
 		return BlockEdit;
 	}
 
-	const EnhancedBlockEdit = function( props ) {
-		const { attributes: { ampLayout }, name } = props;
+	const EnhancedBlockEdit = ( props ) => {
+		const { isSelected, name } = props;
 
-		let inspectorControls;
-
-		if ( 'core/gallery' === name ) {
-			inspectorControls = setUpGalleryInspectorControls( props );
-		} else if ( 'core/image' === name ) {
-			inspectorControls = setUpImageInspectorControls( props );
-		} else if ( MEDIA_BLOCKS.includes( name ) || 0 === name.indexOf( 'core-embed/' ) ) {
-			inspectorControls = setUpInspectorControls( props );
+		if ( isSelected && 'core/image' === name ) {
+			return (
+				<>
+					<BlockEdit { ...props } />
+					<ImageBlockLayoutAttributes { ...props } />
+				</>
+			);
 		}
 
-		// Return just inspector controls in case of 'nodisplay'.
-		if ( ampLayout && 'nodisplay' === ampLayout ) {
-			return [
-				inspectorControls,
-			];
+		if ( isSelected && 'core/gallery' === name ) {
+			return (
+				<>
+					<BlockEdit { ...props } />
+					<GalleryBlockLayoutAttributes { ...props } />
+				</>
+			);
 		}
 
-		return (
-			<>
-				<BlockEdit { ...props } />
-				{ inspectorControls }
-			</>
-		);
+		return <BlockEdit { ...props } />;
 	};
 
 	EnhancedBlockEdit.propTypes = {
 		attributes: PropTypes.shape( {
-			text: PropTypes.string,
-			ampLayout: PropTypes.string,
+			ampLightbox: PropTypes.bool,
```

Comment:
I believe this is missing `ampCarousel`?

```suggestion
			ampCarousel: PropTypes.bool,
			ampLightbox: PropTypes.bool,
```